src/node/mini_miner.cpp

```diff
@@ -0,0 +1,321 @@
+    for (const auto& outpoint : outpoints) {
+        if (const auto ptx{mempool.GetConflictTx(outpoint)}) {
+            // This outpoint is already being spent by another transaction in the mempool.
+            // We assume that the caller wants to replace this transaction (and its descendants).
+            // If the outpoint is from a mempool transaction, we still need to calculate its
+            // ancestors bump fees (added to m_requested_outpoints_by_txid below), but after
+            // removing the to-be-replaced entries. Note that this is only calculating bump fees.
+            // RBF fee rules should be handled separately.
+            m_to_be_replaced.insert(ptx->GetHash());
+            // Remove descendants because they will be replaced as well. This case should be rare
+            // as the wallet won't normally attempt to replace transactions with descendants.
+            CTxMemPool::setEntries descendants;
+            mempool.CalculateDescendants(mempool.GetIter(ptx->GetHash()).value(), descendants);
+            for (const auto& desc_txiter : descendants) {
+                m_to_be_replaced.insert(desc_txiter->GetTx().GetHash());
+            }
+        }
+
+        if (mempool.exists(GenTxid::Txid(outpoint.hash))) {
+            // This UTXO is unconfirmed and in the mempool.
+            auto it = m_requested_outpoints_by_txid.find(outpoint.hash);
+            if (it != m_requested_outpoints_by_txid.end()) {
+                it->second.push_back(outpoint);
+            } else {
+                std::vector<COutPoint> outpoints_of_tx({outpoint});
+                m_requested_outpoints_by_txid.emplace(outpoint.hash, outpoints_of_tx);
+            }
+        } else {
+            // This UTXO is either confirmed or not yet submitted to mempool.
+            // If it's confirmed, no bump fee is required.
+            // If it's not yet submitted, we have no information, so return 0.
+            m_bump_fees.emplace(outpoint, 0);
+        }
+    }
+
+    // No unconfirmed UTXOs, so nothing mempool-related needs to be calculated.
+    if (m_requested_outpoints_by_txid.empty()) return;
+
+    // Calculate the cluster and construct the entry map.
+    std::vector<uint256> txids_needed;
+    for (const auto& [txid, outpoints]: m_requested_outpoints_by_txid) {
+        txids_needed.push_back(txid);
+    }
+    const auto cluster = mempool.CalculateCluster(txids_needed);
+    if (cluster.empty()) {
+        // An empty cluster means that at least one of the transactions is missing from the mempool
+        // (should not be possible given processing above) or DoS limit was hit.
+        m_ready_to_calculate = false;
+        return;
+    }
+
+    // Add every entry to m_entries_by_txid and m_entries, except the ones that will be replaced.
+    for (const auto& txiter : cluster) {
+        if (m_to_be_replaced.find(txiter->GetTx().GetHash()) == m_to_be_replaced.end()) {
```

Comment:
```suggestion
        if (!m_to_be_replaced.count(txiter->GetTx().GetHash())) {
```

Reply:
Can you explain why using `count` is better?